model/messages/dkg.go

```diff
@@ -37,8 +35,22 @@ type PrivDKGMessageOut struct {
 
 // BroadcastDKGMessage is a wrapper around a DKGMessage intended for broadcasting.
 // It contains a signature of the DKGMessage signed with the staking key of the
-// sender.
+// sender. When the DKG contract receives BroadcastDKGMessage' it will attach the
+// NodeID of the sender, we then add this field to the BroadcastDKGMessage when reading broadcast messages.
 type BroadcastDKGMessage struct {
 	DKGMessage
+	Orig      uint64          `json:"-"` // Orig field is set when reading broadcast messages using the NodeID to find the index of the sender in the DKG committee
+	NodeID    flow.Identifier `json:"-"` // NodeID field is added when reading broadcast messages from the DKG contract, this field is ignored when sending broadcast messages
 	Signature crypto.Signature
 }
+
+// PrivateDKGMessage is a wrapper around DKGMessage intended for use when communicating
+// incoming private DKG messages from the messaging engine. This wrapper adds Orig or DKG committee index of the sender
+// which is needed when processing the DKG message in the crypto library.
+// PrivDKGMessageIn component flow:		message_engine -> broker -> controller
+// At the point where the private message reaches the broker, the broker will get the DKG committee index of the sender
+// validate and attach it to an instance of PrivateDKGMessage that will then be forwarded to the controller.
+type PrivateDKGMessage struct {
```

Comment:
- This structure seems to be used only when receiving a private message (not when sending). It is mainly another version of `PrivDKGMessageIn` that substitute the ID with an index. It would be good to have it called something to reflect the "In" aspect, maybe one should be declared in the broker, and the other in the controller, instead of both in model/messages ?

- A different design choice is made for broadcast messages  while it is the same case as Private messages. The struct `BroadcastDKGMessage` is used by both the controller and broker, by having both the ID or index fields. I don't have a preference for either design choice, but it would be good if we keep the same consistent choice for both broadcast and private messages.

Reply:
I decided to add CommitteeMemberIndex to PrivDKGMessageIn and remove the wrapper
https://github.com/onflow/flow-go/pull/2238/commits/df9129e27e41825e65077a23fa471f3f37b239b2